frontend/lib/js/category-trees/actions.js

```diff
@@ -90,3 +93,22 @@ export const loadTree = (datasetId: DatasetIdType, treeId: TreeNodeIdType) => {
       );
   };
 };
+
+export const searchTreesStart = (query: string) =>
+  ({type: SEARCH_TREES_START, payload: { query }});
+export const searchTreesEnd = (query: string, result: []) =>
+  ({type: SEARCH_TREES_END, payload: { query, result }});
+export const searchTreesError = (query: string, err: any) =>
+  defaultError(SEARCH_TREES_ERROR, err, { query });
+
+export const searchTrees = (datasetId: DatasetIdType, query: string) => {
+  return (dispatch: Dispatch) => {
+    dispatch(searchTreesStart(query))
```

Comment:
We should check somewhere if the query is empty (i.e. has been cleared) -- in this case a server request results in an error, and we have a short delay before the unfiltered tree shows.

Reply:
The server request is now not executed on an empty string